lua/neogit/lib/git/repository.lua

```diff
@@ -1,16 +1,8 @@
-local M = {}
-
-local meta = {
-  __index = {},
-}
-
-local modules = { "status", "diff", "stash", "pull", "push", "log", "rebase", "cherry_pick", "merge" }
-for _, m in ipairs(modules) do
-  require("neogit.lib.git." .. m).register(meta.__index)
-end
+local a = require("plenary.async")
```

Comment:
This module was rewritten to act as a singleton, instead of being bound as an instance on the status module. This resulted in (imo) better encapsulation, and let us remove many random imports of the status buffer in different libs just to get a stateful repo. 

It also meant I could extract the state-updating logic from the status buffer, which is a huge win, since that guy needs to get slimmed down and reimplemented using the buffer lib at some point.

Reply:
Yes I saw that and thought the same.

Also neat to move the refresh logic to the repository module.

Reply:
I thought it made a lot more sense to say "Hey repo, refresh yourself" vs "Hey repo... I'm refreshing you"